docs/searching-algorithms/interpolation-search.md

```diff
@@ -0,0 +1,260 @@
+
+  }
+  cout << std::endl;
+
+  //Entering the searching element
+  std::cout << "Enter element to be searched: " << std::endl;
+  cin >> x; //Element to be searched
+  int index = interpolationSearch(arr, n, x);
+  //If element was found
+  if (index != -1)
+    cout << "Element found at index " << index; //Returning the index pf the element
+  else
+    cout << "Element not found."; //Returning -1 as element has not found
+
+  return 0;
+}
+```
+
+#### Sample Input:
+
+```
+5
+1
+7
+4
+9
+2
+4
+```
+
+#### SAMPLE OUTPUT:
+
+Enter size of an array:  
+Enter elements:  
+Array after sorting using default sort is :  
+1 2 4 7 9  
+Enter element to be searched:  
+4  
+Element found at index 2
+
+### Python
+
+```python
+# INTERPOLATION SEARCH
+
+# If the searching element is present in array,
+# then it returns index of it, else returns -1
+
+
+def interpolationSearch(arr, size, search_element):
+    # Assigning indexes
+    low = 0
+    high = size - 1
+    # As array is going to be sorted
+    # an element present in array must be in range defined by corner
+    while low <= high and search_element >=
+    arr[low] and search_element <= arr[high]:
+        if low == high:
+            if arr[low] == search_element:
+                return low
+            return -1
+
+        position = low + int(float(high - low) /
+        (arr[high] - arr[low])*(search_element - arr[low]))
+
+        # Condition of element found
+
+        if arr[position] == search_element:
+            return position
+
+        # If x is larger, x is in upper part
+
+        if arr[position] < search_element:
+            low = position + 1
+        else:
+            # If x is smaller, x is in lower part
+            high = position - 1
+
+    return -1
+
+
+# Starting with the main code
+# Creating array of the size n
+# Size would be taken into the variable n from user
+def main():
+    arr = []
+    size = int(input('Enter the size of array : '))
+
+    # Entering the elements in the array
+
+    print ('Enter elements :')
+    for index in range(0, size):
+        ele = int(input())
+        arr.append(ele)
+    print ('The elements entered are:\n', arr, '\n')
+
+    # Sorting our array
+
+    print ('The sorted array is: ')
+    arr.sort()
+    print (arr, '\n')
+
+    # Entering the searching element
+
+    search_element = int(input('Enter the element to be searched: '))
+
+    index = interpolationSearch(arr, size, search_element)
+
+    # If element was found
+
+    if index != -1:
+        # Returning the index of the element
+        print ('Element found at index', index)
+    else:
+        # Returning -1 as element has not found
+        print ('Element not found')
+main()
+```
+
+#### Sample Input:
+
+```
+5
+1
+5
+7
+2
+9
+7
+```
+
+#### Sample Output:
+
+Enter the size of array :  
+Enter elements :  
+The elements entered are: [1, 5, 7, 2, 9]  
+The sorted array is: [1, 2, 5, 7, 9]  
+Enter the element to be searched:  
+Element found at index 3
```

Comment:
This one is a bit long but worth it. So you can change the file extension to `MDX` instead of `MD`. So the filename would be `interpolation-search.mdx` instead of `interpolation-search.md` 

Once you do that, you would be able to use JSX (React Code) inside Markdown files. The reason we are doing it so that we can stack the languages as per our convenience. 

So currently two implementations in Python and C++ will make it too long. So to shorten that use this syntax: 

```
import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

<Tabs
  defaultValue="js"
  values={[
    { label: 'JavaScript', value: 'js', },
    { label: 'Python', value: 'py', },
    { label: 'Java', value: 'java', },
  ]
}>
<TabItem value="js">

```js
function helloWorld() {
  console.log('Hello, world!');
}```

</TabItem>
<TabItem value="py">

```py
def hello_world():
  print 'Hello, world!'```

</TabItem>
<TabItem value="java">

```java
class HelloWorld {
  public static void main(String args[]) {
    System.out.println("Hello, World");
  }
}```

</TabItem>
</Tabs>
```

This will make the Code look tabbed like this: 

![image](https://user-images.githubusercontent.com/47351025/113996334-b548af80-9874-11eb-88a3-f3c94f3e7f46.png)

Do note that use the end backticks for the individual code at the end of the code line not beside it. I used it in this example so that the Github Markdown doesn't break. 

Check the official [docs](https://docusaurus.io/docs/markdown-features/code-blocks#multi-language-support-code-blocks) to know more.

Also ensure that both C++ and Python Code has the same logging statements. We can keep only one sample I/O for our purpose. And to note that please use triple-backticks for both Input and Output blocks